tests/BoreholeControllerTest.cs

```diff
@@ -92,6 +102,14 @@ public async Task Copy()
             // Delete borehole copy
             if (copiedBorehole != null)
             {
+                var stratigraphiesToRemove = copiedBorehole.Stratigraphies;
+                var layersToRemove = stratigraphiesToRemove.SelectMany(s => s.Layers);
+                var lithologicalDescriptionsToRemove = stratigraphiesToRemove.SelectMany(s => s.LithologicalDescriptions);
+                var faciesDescriptionsToRemove = stratigraphiesToRemove.SelectMany(s => s.FaciesDescriptions);
+                context.Layers.RemoveRange(layersToRemove);
+                context.LithologicalDescriptions.RemoveRange(lithologicalDescriptionsToRemove);
+                context.FaciesDescriptions.RemoveRange(faciesDescriptionsToRemove);
+                context.Stratigraphies.RemoveRange(stratigraphiesToRemove);
```

Comment:
Q: Brauchts das? Wird das nicht per CASCADE DELETE automatisch abgeräumt, wenn man eine Bohrung löscht?

Reply:
Ja das war die Meinung, Patrick und ich haben es gemeinsam angeschaut und nicht herausgefunden, warum das Cascade-delete nicht das macht was es soll. In den Borehole Controller Tests sind zweit Tests die eine Kopie erstellen und einer der beiden räumte bisher schon so ab wie hier, der andere einfach "nur" über das Löschen des Boreholes mit Cascade-Delete.  Das war leider kaputt und ich habe mich der Einfachheit halber dazu entschieden, es hier so zu machen. 
Wenn man direkt auf der DB oder im Client löscht scheind das Cascade-Delete aber zu funktionieren.

Reply:
Alles klar, dann müssen wir das separat anschauen. Vermutlich fehlt uns hier eine Konfiguration oder etwas stimmt nicht, wenn man alle Entities im Tracker hat. Siehe auch https://learn.microsoft.com/en-us/ef/core/saving/cascade-delete#where-cascading-behaviors-happen.